h2o-admissibleml/src/main/java/hex/Infogram/InfogramModel.java

```diff
@@ -0,0 +1,330 @@
+      Field[] algoParams = Model.Parameters.class.getDeclaredFields();
+      Field algoField;
+      for (Field oneField : algoParams) {
+        try {
+          String fieldName = oneField.getName();
+          algoField = this.getClass().getField(fieldName);
+          if (excludeList.size() == 0 || !excludeList.contains(fieldName)) {
+            algoField.set(_infogram_algorithm_parameters, oneField.get(this));
+          }
+        } catch (IllegalAccessException | NoSuchFieldException e) { // suppress error printing.  Only care about fields that are accessible
+          ;
+        }
+      }
+    }
+  }
+
+  public static class InfogramModelOutput extends Model.Output {
+    public double[] _admissible_cmi;  // conditional info for admissible features in _admissible_features
+    public double[] _admissible_cmi_raw;  // conditional info for admissible features in _admissible_features raw
+    public double[] _admissible_relevance;  // varimp values for admissible features in _admissible_features
+    public String[] _admissible_features; // predictors chosen that exceeds both conditional_info and varimp thresholds
+    public double[] _admissible_index;  // store normalized distance from 0,0 corner of infogram plot from 0 to 1
+    public double[] _admissible; // 0 if predictor is admissible and 1 otherwise
+    public DistributionFamily _distribution;
+    public double[] _cmi_raw; // cmi before normalization and for all predictors
+    public double[] _cmi; // normalized cmi
+    public String[] _all_predictor_names;
+    public double[] _relevance; // variable importance for all predictors
+    public String _relevance_cmi_key;
+    public Key<Frame> _relCmiKey;
+    public String[] _topKFeatures;
+
+    @Override
+    public ModelCategory getModelCategory() {
+      if (bernoulli.equals(_distribution)) {
+        return ModelCategory.Binomial;
+      } else if (multinomial.equals(_distribution)) {
+        return ModelCategory.Multinomial;
+      } else if (ordinal.equals(_distribution)) {
+        return ModelCategory.Ordinal;
+      }
+      throw new IllegalArgumentException("Infogram currently only support binomial and multinomial classification");
+    }
+
+    public void setDistribution(DistributionFamily distribution) {
+      _distribution = distribution;
+    }
+    
+    public InfogramModelOutput(Infogram b) {
+      super(b);
+      if (glm.equals(b._parms._algorithm)) {
+        if (binomial.equals(((GLMModel.GLMParameters) b._parms._infogram_algorithm_parameters)._family))
+          _distribution = bernoulli;
+        else if (multinomial.equals(((GLMModel.GLMParameters) b._parms._infogram_algorithm_parameters)._family))
+          _distribution = multinomial;
+        else if (ordinal.equals(((GLMModel.GLMParameters) b._parms._infogram_algorithm_parameters)._family))
+          _distribution = ordinal;
+      } else {
+        _distribution = b._parms._infogram_algorithm_parameters._distribution;
+      }
```

Comment:
I feel like I've seen this logic in several places already, can't we just create and share utility functions for this so that we don't have to see this again?
I'd suggest a couple of static utility methods in `GLM` once and for all:
```java
public static final DistributionFamily familyToDistribution(Family family) { … }

public static final Family distributionToFamily(DistributionFamily distribution) { … }
```
I'd suggest to start extracting [this](https://github.com/h2oai/h2o-3/blob/master/h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java#L353) to the first static method in GLM and use it here.

Reply:
ok, maybe I already mentioned this in a previous review as I see that you created `DistributionUtils`, so you can use it here now.

Reply:
we could also eventually clean-up GLM and switch it to "DistributionFamily" and keep "Family" only as a backwards compatiblity option

Reply:
good idea, but probably for another PR

Reply:
Good catch.  Done.